### PR TITLE
CODE-3257: fix pcgen.sh a bit

### PR DESCRIPTION
**Pull request by grimreaper**
- We already cd to correct directory at the top of the file. This one is
extraneous and has incorrect quoting
- remove broken '2> /dev/null' in conditional
- add quotes

### CODE REVIEW

File: code/pcgen.sh [hunk 1]
```diff
@@ -82,5 +82,5 @@ done
 #     -Dpcgen.filter=/path/to/filter.ini
 #     -Dpcgen.options=/path/to/options.ini
 
-cd $(dirname $(readlink -f $0))
+# shellcheck disable=SC2086
```
Comment (LegacyKing) [inline code/pcgen.sh]: Why is this commented out? Doesn't follow the principles of clean code.

Comment (grimreaper) [reply code/pcgen.sh]: This is a "lint supression". It is supposed to be a comment. It tells the tool 'shellcheck' not to report `$javaargs` as unquoted. we want it as such in order to expand.

Comment (LegacyKing) [reply code/pcgen.sh]: Gotcha.

Comment (LegacyKing) [reply code/pcgen.sh]: I should install a linux build to properly test these linux only issues...

**Pull request merged**